clients/destination.go

```diff
@@ -193,6 +195,26 @@ func (c *DestinationClient) Initialize(ctx context.Context, spec specs.Destinati
 	return nil
 }
 
+func (c *DestinationClient) Validate(ctx context.Context, spec specs.Destination) (warnings, errors []string, err error) {
+	b, err := json.Marshal(spec)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal destination spec: %w", err)
+	}
+	resp, err := c.pbClient.Validate(ctx, &pb.ValidateDestination_Request{
+		Spec: b,
+	})
+	if err != nil {
+		st, ok := status.FromError(err)
+		if ok && st.Code() == codes.Unimplemented {
+			// Backwards-compatibility with older plugin versions that don't support Validate().
+			// In this case, we only return one warning: that the plugin should be updated.
+			return []string{"the version of this plugin is outdated and should be updated"}, nil, nil
```

Comment:
Nit: Maybe temper the wording a bit. 
Some advanced features might not be supported by the current version of the CLI, and might lead to breaks in future. Please update using (this link)

Reply:
Thaks, good suggestion - I was struggling to find the right wording here :)